File: ktink/maxProfit.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int index = 0;
        int size = prices.size() - 1;
        int x = 0;
        int cnt = 0;
        for (; index < size; index ++) {
            x = prices[index + 1] -  prices[index];
            if (x > 0) {
                cnt += x;
            }
        }
        return cnt;
    }
};

File: bert/2022-01-04/maxProfit.go
package main

func main() {

}

func maxProfit(prices []int) int {
	var sum int
	for i := 0; i < len(prices)-1; i++ {
		if prices[i+1]-prices[i] > 0 {
			sum += prices[i+1] - prices[i]
		}
	}
	return sum
}
